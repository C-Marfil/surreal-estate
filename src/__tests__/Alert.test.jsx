import React from "react";
import { render } from "@testing-library/react";
import Alert from "../components/Alert";

describe("Alert component", () => {
  it("displays an error message", () => {
    const { getByText } = render(<Alert message="Error!" />);

    expect(getByText(/Error/).textContent).toBe("Error!");
  });

  it("displays a success message", () => {
    const { getByText } = render(<Alert message="Success!" />);

    expect(getByText(/Success/).textContent).toBe("Success!");
  });

  it("doesn't render the alert when message is falsy", () => {
    const { asFragment } = render(<Alert message="" />);
    const alert = asFragment();

    expect(alert).toMatchSnapshot();
  });

  it("tests it renders when successful", () => {
    const { asFragment } = render(<Alert message="Success!" success />);
    const alert = asFragment();

    expect(alert).toMatchSnapshot();
  });

  it("tests it renders when error", () => {
    const { asFragment } = render(<Alert message="Error!" success={false} />);
    const alert = asFragment();

    expect(alert).toMatchSnapshot();
  });
});
