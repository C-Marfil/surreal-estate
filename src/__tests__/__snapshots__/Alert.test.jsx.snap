// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component doesn't render the alert when message is falsy 1`] = `<DocumentFragment />`;

exports[`Alert component tests it renders when error 1`] = `
<DocumentFragment>
  <p
    class="alert alert-error"
  >
    Error!
  </p>
</DocumentFragment>
`;

exports[`Alert component tests it renders when successful 1`] = `
<DocumentFragment>
  <p
    class="alert alert-success"
  >
    Success!
  </p>
</DocumentFragment>
`;
